deepchem/feat/materials_featurizers.py

```diff
@@ -0,0 +1,250 @@
+  of atoms in a crystal structure. The chemical fingerprint is a 
+  vector of these statistics. For a full list of properties and statistics,
+  see ElementProperty(data_source).feature_labels().
+
+  This featurizer requires the optional dependencies pymatgen and
+  matminer. It may be useful when only crystal compositions are available
+  (and not 3D coordinates).
+
+  References are given for each data source:
+    MagPie data: Ward, L. et al. npj Comput Mater 2, 16028 (2016).
+    https://doi.org/10.1038/npjcompumats.2016.28
+
+    Deml data: Deml, A. et al. Physical Review B 93, 085142 (2016).
+    10.1103/PhysRevB.93.085142
+
+    Matminer: Ward, L. et al. Comput. Mater. Sci. 152, 60-69 (2018).
+
+    Pymatgen: Ong, S.P. et al. Comput. Mater. Sci. 68, 314-319 (2013). 
+
+  """
+
+  def __init__(self, data_source='matminer'):
+    """
+    Parameters
+    ----------
+    data_source : {"matminer", "magpie", "deml"}
+      Source for element property data.
+
+    """
+
+    self.data_source = data_source
+
+  def _featurize(self, comp):
+    """
+    Calculate chemical fingerprint from crystal composition.
+
+    Parameters
+    ----------
+    comp : str
+      Reduced formula of crystal.
+
+    """
+
+    from pymatgen import Composition
+    from matminer.featurizers.composition import ElementProperty
+
+    # Get pymatgen Composition object
+    c = Composition(comp)
+
+    ep = ElementProperty.from_preset(self.data_source)
+
+    try:
+      feats = ep.featurize(c)
+    except:
+      feats = []
+
+    return feats
+
+
+class SineCoulombMatrix(Featurizer):
+  """
+  Calculate sine Coulomb matrix for crystals.
+
+  A variant of Coulomb matrix for periodic crystals, based on 
+  Faber et al. Inter. J. Quantum Chem. 115, 16, (2015).
+
+  The sine Coulomb matrix is identical to the Coulomb matrix, except
+  that the inverse distance function is replaced by the inverse of
+  sin**2 of the vector between sites which are periodic in the 
+  dimensions of the crystal lattice.
+
+  Features are flattened into a vector of matrix eigenvalues by default
+  for ML-readiness. To ensure that all feature vectors are equal
+  length, the maximum number of atoms (eigenvalues) in the input
+  dataset must be specified.
+
+  This featurizer requires the optional dependency pymatgen. It may be
+  useful when crystal structures with 3D coordinates are available.
+
+  """
+
+  def __init__(self, max_atoms, eig=True):
+    """
+    Parameters
+    ----------
+    max_atoms : int
+      Maximum number of atoms for any crystal in the dataset. Used to
+      pad the Coulomb matrix.
+    eig : bool (default True)
+      Return flattened vector of matrix eigenvalues.
+
+    """
+
+    self.max_atoms = int(max_atoms)
+    self.eig = eig
+
+  def _featurize(self, struct):
+    """
+    Calculate sine Coulomb matrix from pymatgen structure.
+
+    Parameters
+    ----------
+    struct : dict
+      pymatgen structure dictionary
```

Comment:
Can you add more details on what a `pymatgen structure dictionary` is? Perhaps a reference to some of the pymatgen documentation specifying what it is would be useful.